src/repl-frontend/scala/tools/nsc/interpreter/jline/Reader.scala

```diff
@@ -220,38 +220,43 @@ class Completion(delegate: shell.Completion) extends shell.Completion with Compl
   // REPL Completion
   def complete(buffer: String, cursor: Int): shell.CompletionResult = delegate.complete(buffer, cursor)
 
-  def reset(): Unit = delegate.reset()
-
   // JLine Completer
   def complete(lineReader: LineReader, parsedLine: ParsedLine, newCandidates: JList[Candidate]): Unit = {
-    def candidateForResult(s: String): Candidate = {
-      val value = s
-      val displayed = s
+    def candidateForResult(cc: CompletionCandidate): Candidate = {
+      val value = cc.defString
+      val displayed = cc.defString + (cc.arity match {
+        case CompletionCandidate.Nullary => ""
+        case CompletionCandidate.Nilary => "()"
+        case _ => "("
```

Comment:
would be nice if `(` / `()` were also inserted after selecting a candidate.

Reply:
JLine does support that (they call it "suffix"). I've recorded the suggestion at https://github.com/scala/scala-dev/issues/680#issuecomment-617797030